### PR TITLE
Json output

### PR DESCRIPTION
**Pull request by ehildenb**
@dwightguth this introduces the Json parser, and adds a `KPrint` test which exercises both the `JSON` and `BINARY` parser/unparser. I still need to figure out how to run the `KAST` parser.

### CODE REVIEW

**Comment by ehildenb** [comment]
Blocked on merging the #117 .

**Comment by ehildenb** [comment]
@dwightguth ready for review.

**Comment by ehildenb** [comment]
Jenkins: test this please

File: kernel/src/main/java/org/kframework/parser/json/JsonParser.java
Comment (dwightguth) [inline kernel/src/main/java/org/kframework/parser/json/JsonParser.java]: This should really catch UnsupportedEncodingException and rethrow it as an AssertionError because otherwise it pollutes the type of the function with a throws clause even though UTF-8 is a required encoding on all conforming implementations of the JVM.

File: kernel/src/main/java/org/kframework/parser/json/JsonParser.java
Comment (dwightguth) [inline kernel/src/main/java/org/kframework/parser/json/JsonParser.java]: This does not correctly handle the case when the klabel is a variable.

File: kernel/src/main/java/org/kframework/parser/json/JsonParser.java
Comment (dwightguth) [inline kernel/src/main/java/org/kframework/parser/json/JsonParser.java]: If you want to be able to serialize and deserialize arbitrary rules, you need to also include a case for KAs.

File: kernel/src/main/java/org/kframework/parser/json/JsonParser.java
Comment (dwightguth) [inline kernel/src/main/java/org/kframework/parser/json/JsonParser.java]: This also does not correctly handle variables.

File: kernel/src/main/java/org/kframework/unparser/ToJson.java
Comment (dwightguth) [inline kernel/src/main/java/org/kframework/unparser/ToJson.java:129]: This also needs a variable field.

File: kernel/src/main/java/org/kframework/unparser/ToJson.java
Comment (dwightguth) [inline kernel/src/main/java/org/kframework/unparser/ToJson.java]: Once again you are missing the case for KAs.

**Review by dwightguth — CHANGES_REQUESTED** [review]
Looks pretty good but there are some tricky corner cases.

**Comment by ehildenb** [comment]
@dwightguth I believe I've addressed your concerns:

-   I've fixed the `UnsupportedEncodingException => AssertionError`.
-   I've added both Json serializer/deserializer for `KAs` case.
-   I've handled the cases where a `KLabel` can also be a `KVariable`.

**Review by dwightguth — APPROVED** [review]

**Comment by ehildenb** [comment]
@dwightguth feel free to merge when you feel ready, I still do not have merge permissions.

**Pull request merged**
